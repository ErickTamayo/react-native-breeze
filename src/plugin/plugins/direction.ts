import { PluginFunction } from "./types";

export type PluginGroups = { writing: "inherit" | "ltr" | "rtl" };

export const pattern = /^direction-(?<writing>inherit|ltr|rtl)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { writing } = groups;
  return { direction: writing };
};
