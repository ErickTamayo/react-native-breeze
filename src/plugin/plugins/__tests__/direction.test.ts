import {
  shouldEvaluateTheCorrectPatternTest,
  shouldParseCorrectlyTest,
} from "../../utils/tests";
import { pattern, plugin } from "../direction";

describe("direction", () => {
  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch: ["direction-inherit", "direction-ltr", "direction-rtl"],
    shouldNotMatch: ["direction", "unknown"],
  });

  shouldParseCorrectlyTest(pattern, plugin, [
    ["direction-inherit", { direction: "inherit" }],
    ["direction-ltr", { direction: "ltr" }],
    ["direction-rtl", { direction: "rtl" }],
  ]);
});
