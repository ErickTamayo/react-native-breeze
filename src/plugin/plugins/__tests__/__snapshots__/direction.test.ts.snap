// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direction Should parse (direction-inherit) correctly 1`] = `
Object {
  "direction": "inherit",
}
`;

exports[`direction Should parse (direction-ltr) correctly 1`] = `
Object {
  "direction": "ltr",
}
`;

exports[`direction Should parse (direction-rtl) correctly 1`] = `
Object {
  "direction": "rtl",
}
`;
